engine/classes/ElggMenuItem.php

```diff
@@ -396,6 +399,39 @@ public function addLinkClass($class) {
 	}
 
 	/**
+	 * Set required AMD modules
+	 *
+	 * @param string[]|string $modules One or more required AMD modules
+	 * @return void
+	 */
+	public function setRequire($modules) {
+		$this->data['require'] = (array) $modules;
+	}
+
+	/**
+	 * Get required AMD modules
+	 *
+	 * @return string[]
+	 */
+	public function getRequire() {
+		$modules = (array) $this->data['require'];
+		return array_filter($modules, function($m) {
```

Comment:
We should validate in __construct() too? Generally we shouldn't invite bad state into an object on assumption we'll clean it later. M